src/postgres/abstract/AbstractPostgresClient.ts

```diff
@@ -0,0 +1,82 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { PostgreSQLManagementClient } from "@azure/arm-postgresql";
+import { PostgreSQLFlexibleManagementClient } from "@azure/arm-postgresql-flexible";
+import * as msRest from "@azure/ms-rest-js";
+import { IMinimumServiceClientOptions } from "vscode-azureextensionui";
+import { asAbstractDatabase, asFlexibleParameters, asSingleParameters, flexibleAsAbstractServer, singleAsAbstractServer } from "./maps";
+import * as Models from "./models";
+
+export class AbstractPostgresClient {
+    private postgresFlexibleClient: PostgreSQLFlexibleManagementClient;
+    private postgresSingleClient: PostgreSQLManagementClient;
+
+    constructor(credentials: msRest.ServiceClientCredentials, subscriptionId: string, options?: IMinimumServiceClientOptions) {
+        this.postgresFlexibleClient = new PostgreSQLFlexibleManagementClient(credentials, subscriptionId, options);
+        this.postgresSingleClient = new PostgreSQLManagementClient(credentials, subscriptionId, options);
+    }
+
+    async checkNameAvailability(serverType: Models.PostgresServerType, name: string) : Promise<Models.AbstractNameAvailability> {
+        switch (serverType){
+            case Models.PostgresServerType.Flexible:
+                return this.postgresFlexibleClient.checkNameAvailability.execute({name: name, type: "Microsoft.DBforPostgreSQL"});
+            case Models.PostgresServerType.Single:
+                return this.postgresSingleClient.checkNameAvailability.execute({name: name, type: "Microsoft.DBforPostgreSQL"});
+            default:
+                throw new Error("Service not implemented.");
+        }
+    }
+
+    async createOrUpdateFirewallRule(serverType: Models.PostgresServerType, resourceGroup: string, name: string, ruleName: string, rule: Models.AbstractFirewallRule): Promise<Models.AbstractFirewallRule> {
+        switch (serverType){
+            case Models.PostgresServerType.Flexible:
+                return this.postgresFlexibleClient.firewallRules.createOrUpdate(resourceGroup, name, ruleName, rule);
+            case Models.PostgresServerType.Single:
+                return this.postgresSingleClient.firewallRules.createOrUpdate(resourceGroup, name, ruleName, rule);
+            default:
+                throw new Error("Service not implemented.");
+        }
+    }
+
+    async createServer(serverType: Models.PostgresServerType, resourceGroup: string, name: string, parameters: Models.AbstractServerCreate): Promise<Models.PostgresAbstractServer> {
```

Comment:
`createServer` and `listAllServers` are the only two methods on this class that aren't simple copy/paste and thus the only reason we _might_ need this class still. However, since each method is only used once, I don't think there's much value to putting them in a separate class like this. The true value of abstracting it away would be if we were calling the methods multiple times, which I don't see happening. I say we go with my `createAbstractPostgresClient` function above and move this code directly into `PostgresServerCreateStep`